### DISCUSSION TITLE
How to describe source term in diffusion function?

### DISCUSSION THREAD

**Discussion by lipxisi** [idaholab/moose#23783] [Q&A General] [answered]
I have a source term S(r)=1x10e+14cos((pi r)/2R), How to describe it in kernel?
Thans for your answer.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Use this to create a function
https://mooseframework.inl.gov/source/functions/MooseParsedFunction.html
then this to make it a source term
https://mooseframework.inl.gov/source/kernels/ADBodyForce.html

Depending on whether you are solving this in 3D or 2D RZ, you'll want to write r as x^2 + y^2 or just x

Guillaume

**Reply by lipxisi** [reply]
Hello Guillaume. Thanks for your answer.  Are the following code are right if I describe the source in 3D?
[Functions]
  [./sourceterm]
      type=ParsedFunction
      expression=(10e+14)*cos(pi*(x^2+y^2)/2R)
      symbol_names='R'
      symbol_values='1'
  []
[]

Shouldn`t the x^2 + y^2 be r^2?

**Reply by GiudGiud** [reply]
oh yes you need to take the square root to make r

**Reply by lipxisi** [reply]
Hello

Can the distribution of source items be printed separately like output of mesh? I want to see if the source item is correct.
Thanks

**Reply by GiudGiud** [reply]
You can 

you ll have to use the vector tag parameters to save the contribution to the residual to a new system vector

then you can use a TagVectorAux (or some similar name) to retrieve it in Au auxiliary variable
